src/GslibIO.jl

```diff
@@ -155,4 +196,58 @@ function save(file::File{format"GSLIB"}, sdata::AbstractData)
        propnames=vars)
 end
 
+# low level function for saving `data` to a legacy GSLIB format using `varnames` as variable names
+function save_legacy(filename::AbstractString, data::AbstractMatrix, varnames::NTuple, header::AbstractString, na)
+  @assert size(data, 2) == length(varnames) "Invalid data for the specified variable names"
+  nvars = size(data, 2)
+
+  open(filename, "w") do f
+    write(f, "$header\n")
+
+    write(f, "$nvars\n")
+    for v in varnames
+      write(f, "$v\n")
+    end
+
+    # handle missing values
+    replace!(data, NaN=>na)
```

Comment:
We shouldn't modify the data in place given that it is passed to the function. Also, try to move this handling of missing values to the beginning of the function before opening the file.

Reply:
Which name do you suggest for the local variable? datawithnan?

Reply:
perhaps `datana` ?